minefield_game/statePattern/EndGameState.cpp

```diff
@@ -0,0 +1,25 @@
+#include "GameContext.h"
+#include "EndGameState.h"
+#include "PlacementState.h"
+#include "../board_header.h"
```

Comment:
#must not use relative paths this way. Instead, supply the proper include path to the build command. I.e. `clang -I. *.cpp -o minefield.exe`. The `-I` option is used to specify an include path (you can specify many by repeating the argument, i.e. `-Ifolder1 -Ifolder2`, etc.), and `.` points to current directory. Then you can remove the `../`, and it'd be better if you use `<>` in that case (i.e. `#include <board.h>`)